### PR TITLE
fix(bar-label): allow fill function accessing label bound to support checking overlapping

### PR DESCRIPTION
**Pull request by quanho**
There is a need to decide the color of a label depending on if the label is overlapping with other objects. To be able to do that, in the fill color function, we need to know the label bound to support checking overlapping between the label and other objects.

### CODE REVIEW

File: packages/picasso.js/src/core/chart-components/labels/strategies/bars.js [hunk 1]
```diff
@@ -308,13 +306,16 @@ export function placeInBars(
         });
 
         if (label) {
+          const textBounds = approxTextBounds(label, measured, isRotated, bounds, placement.padding);
+          fill = typeof placement.fill === 'function' ? placement.fill(arg, i, textBounds) : placement.fill;
```
Comment (cbt1) [inline packages/picasso.js/src/core/chart-components/labels/strategies/bars.js]: Don't like adding `textBounds` as a 3rd argument. A better solution IMO would be to extend the first argument with the `textBounds`

File: packages/picasso.js/src/core/chart-components/labels/strategies/bars.js [hunk 2]
```diff
@@ -308,13 +306,16 @@ export function placeInBars(
...
+          label.fill = fill;
           if (typeof linkData !== 'undefined') {
             label.data = linkData;
           }
           if (typeof placement.background === 'object') {
             label.backgroundColor =
               typeof placement.background.fill === 'function'
-                ? placement.background.fill(arg, i)
+                ? placement.background.fill(arg, i, textBounds)
```
Comment (cbt1) [inline packages/picasso.js/src/core/chart-components/labels/strategies/bars.js]: Same comment as above regarding 3rd argument

Comment (quanho) [reply packages/picasso.js/src/core/chart-components/labels/strategies/bars.js]: Fixed!

File: packages/picasso.js/src/core/chart-components/labels/strategies/bars.js (hunk 1 above)
Comment (quanho) [reply packages/picasso.js/src/core/chart-components/labels/strategies/bars.js]: Fixed!

**Pull request merged**